### PR TITLE
implemented registration and login for the user

### PR DESCRIPTION
**Pull request by mark-nakachon**
### Checklist:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pull) for the same update/change?
* [x] Have you successfully ran tests with your changes locally?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### Issues Solved
Now user can be able to log in and signup
Fixes #{90}

### CODE REVIEW

**Comment by mark-nakachon** [comment]
please review this and merge this pr

**Comment by mark-nakachon** [comment]
Please review and merge this PR make it fast please

File: server/index.js [hunk 1]
```diff
@@ -57,7 +107,37 @@ app.post('/api/upload', function(req, res) {
...
+		firstName: req.body.firstName,
+		lastName: req.body.lastName,
+		email: req.body.email,
+		password: req.body.password
+	});
+	bcrypt.genSalt(10, (error, salt) => {
+		bcrypt.hash(newUser.password, salt, (err, hash) => {
+			newUser.password = hash;
+			newUser
+				.save()
```
Comment (rossoskull) [inline server/index.js]: Before saving the account here, you should verify it once in the backend too. Add a regex check for email, and checks for the password too. Also, intimate the user with any errors that come forward.

Comment (mark-nakachon) [reply server/index.js]: Woking on it

**Comment by mark-nakachon** [comment]
Added form validation please review this

**Comment by mark-nakachon** [comment]
Added changes requested by you now please merge this

File: views/pages/signup.ejs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>Upload File</title>
```
Comment (rossoskull) [inline views/pages/signup.ejs]: This should be Sign Up

File: views/pages/login.ejs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>Upload File</title>
```
Comment (rossoskull) [inline views/pages/login.ejs]: This should be Log In

File: views/pages/signup.ejs (hunk 2 above)
Comment (mark-nakachon) [reply views/pages/signup.ejs]: Fixed this issue

File: views/pages/login.ejs (hunk 3 above)
Comment (mark-nakachon) [reply views/pages/login.ejs]: Fixed this issue

**Comment by mark-nakachon** [comment]
Fixed the issues

**Comment by mark-nakachon** [comment]
@aashutoshrathi fixed issues requested by you

**Comment by aashutoshrathi** [comment]
Merging this, for now, make sure you do another PR for improving form UI.

@rossoskull Update this in the scoresheet.

**Comment by mark-nakachon** [comment]
Sure I will do this is just only prototyping don't bother about UI I will fix it later

**Pull request merged**